ray-operator/controllers/ray/common/pod.go

```diff
@@ -457,9 +578,19 @@ func concatenateContainerCommand(nodeType rayiov1alpha1.RayNodeType, rayStartPar
 
 	switch nodeType {
 	case rayiov1alpha1.HeadNode:
-		return fmt.Sprintf("ulimit -n 65536; ray start --head %s", convertParamMap(rayStartParams))
+		if needRedisInstalled {
+			return fmt.Sprintf("/home/ray/anaconda3/bin/pip install redis; "+
+				"ulimit -n 65536; ray start --head %s", convertParamMap(rayStartParams))
+		} else {
+			return fmt.Sprintf("ulimit -n 65536; ray start --head %s", convertParamMap(rayStartParams))
+		}
 	case rayiov1alpha1.WorkerNode:
-		return fmt.Sprintf("ulimit -n 65536; ray start %s", convertParamMap(rayStartParams))
+		if needRedisInstalled {
+			return fmt.Sprintf("/home/ray/anaconda3/bin/pip install redis; "+
```

Comment:
I remember last week's comment said anyscale will install redis in the image?

Reply:
We can get that in before this PR is merged.

Just confirming -- Is it correct that the new functionality introduced in this PR is only supposed to be compatible with Ray nightly / 2.0.0?

Reply:
yes, I filed a bug https://github.com/ray-project/ray/issues/26427

@iycheng may take a look at this.

Reply:
https://github.com/ray-project/ray/pull/26446

Reply:
I will remove this part after https://github.com/ray-project/ray/pull/26446 is merged and available in nightly build.